# Começando em Python...

O objetivo desta aula bem com sua atividade é apresentar algumas características do Python e, neste primeiro momento, algumas funcionalidades básicas e bibliotecas importantes

## 1º Exemplo: O clássico "Olá Mundo!"
 

In [1]:
#Para escrever algo em tela, após compilação, basta usar o comando 'print', como na sintaxe abaixo
print("Olá mundo")

# Comentários iniciam-se com hashtag e não precisam ser fechados
# Comentários não aparecem na saída do código após o mesmo ser compilado e também não tem quebra de linha
# Comentários também podem ser inseridos ao final de uma linha de comando

print("olá mundo, de novo")   #teste de comentário



Olá mundo
olá mundo, de novo


## 2º Exemplo: Variáveis

Variáveis são nomes associados a objetos em particular. Elas podem ser uma referências, um ponteiro ou um endereço. 

In [2]:
mensagem = "nossa aula de teste"

print(mensagem)
print(type(mensagem))

"""
Este bloco de comentários aparece após a compilação do código
"""

nossa aula de teste
<class 'str'>


'\nEste bloco de comentários aparece após a compilação do código\n'

O bom senso indica que o nome das variáveis tenham relação com o que representam e não sejam muito longas, para evitar eventuais problemas.

### Importante
Nomes de variáveis em Python aceitam letras maiúsculas, minúsculas (Case Sensitive) além de números e o símbolo de subscrito ( _ ).

As variáveis podem apresentar termos compostos, com letras minúsculas, separadas por subscrito, o que é conhecido com snake_case.

Também é possível combinar letras maiúsculas e minúsculas, que denomina-se PacalCase.

## Palavras chave

Algumas palavras são de uso exclusivo da linguagem (syntax) e não podem ser utilizadas como nome de variáveis. Para saber quais estas palavras, basta digitar o comando help("keywords")

In [3]:
help("keywords")


Here is a list of the Python keywords.  Enter any keyword to get more help.

False               class               from                or
None                continue            global              pass
True                def                 if                  raise
and                 del                 import              return
as                  elif                in                  try
assert              else                is                  while
async               except              lambda              with
await               finally             nonlocal            yield
break               for                 not                 



## Ainda mais sobre variáveis
Em relação a números podemos utilizar Inteiros, Pontos flutuantes (números com casas decimais) e números complexos.

### Importante:
Em Python as casas decimais de um número são separadas por ponto!

In [4]:
# Tipos de variáveis
# int
a = 1
b = 2
print(a+b)
print(type(a))
# float
c = 1.0
d = 1.1
print(c+d)
type(c)
x = complex(1,2)
print(x)
print(x+a)

3
<class 'int'>
2.1
(1+2j)
(2+2j)


## Operadores

In [5]:
#Soma
5 + 3

#Subtração
5-3

#Multiplicação
5*3

#Divisão
5/3

#Parte inteira da divisão
5//3

#Resto da divisão
5%3

#Potência
5**3

125

## Operadores Booleanos

Operadores booleanos são implementados como uma subclasse de inteiros, com apenas dois valores possíveis em Python "True" ou "False".
Estes valores são úteis na comparação de lógica de valores (maior que >, menor que < ou igual ==) ou funções de predicado.

In [6]:
2< 5

4 > 10

4 <= 3

3>=3

5==6

6!=9  # o sinal != significa diferente

True

## Strings

Strings são fragmentos de textos ou caracteres e podem ser definidos usando aspas simples, duplas ou triplas.


In [7]:
#Aspas Simples
mensagem = 'Hello world'
mensagem

'Hello world'

In [8]:
#Aspas Duplas
mensagem = "Hello world"
mensagem

'Hello world'

In [9]:
#Aspas Triplas
mensagem = """Hello world"""
mensagem

'Hello world'

In [10]:
# Juntando expressões
"hello" + " " + "world"

'hello world'

In [11]:
#Maiúsculas
"hello world".upper()

'HELLO WORLD'

In [12]:
#Minúsculas
"HELLO WORLD".lower()

'hello world'

In [13]:
#Retirando strings de uma expressão

boas_vindas = "Hello world"
boas_vindas[0:5]    # Importante que o contador deve começar em 0 (zero)

'Hello'

## Vamos a um pequeno código...

In [14]:
tomou_cafe = False # True
print(type(tomou_cafe))
if tomou_cafe:
    mensagem = "Bom dia, turma querida!"
else:
    mensagem = "Vamos começar essa aula de uma vez!"
print(mensagem)

<class 'bool'>
Vamos começar essa aula de uma vez!


In [15]:
## Definição de função em Python

def escrever_mensagem(mensagem=""):
    
    # Este comentário gera o help da função
    """
    Constrói a mensagem do professor Felipe
    RETURNS: str
    """
    mensagem = "Prof. Felipe disse: " + mensagem
    return mensagem
escrever_mensagem()
print()
print(escrever_mensagem('A função retorna este valor'))


Prof. Felipe disse: A função retorna este valor


In [16]:
help(escrever_mensagem)

Help on function escrever_mensagem in module __main__:

escrever_mensagem(mensagem='')
    Constrói a mensagem do professor Felipe
    RETURNS: str



## Voltando à tabela de salários

Rode os códigos abaixo, considerando o arquivo TabelaSalarial.csv e responda às perguntas ao final.

O prazo final para entrega desta atividade é dia 14 de agosto, quarta-feira, às 23h59


In [17]:
import pandas as pd
dados = pd.read_csv('TabelaSalarial.csv')
dados.head()

N Estado Civil   Grau de Instrução  N de Filhos  Salario (x Sal Min)  \
0  1     solteiro  ensino fundamental          NaN                 4.00   
1  2       casado  ensino fundamental          1.0                 4.56   
2  3       casado  ensino fundamental          2.0                 5.25   
3  4     solteiro        ensino médio          NaN                 5.73   
4  5     solteiro  ensino fundamental          NaN                 6.26   

   Idade (em anos) Região de Procedência  
0               26              interior  
1               32               capital  
2               36               capital  
3               20                 outra  
4               40                 outra

In [18]:
import matplotlib
dados.describe()

N  N de Filhos  Salario (x Sal Min)  Idade (em anos)
count  36.000000    20.000000            36.000000        36.000000
mean   18.500000     1.650000            11.122222        34.583333
std    10.535654     1.268028             4.587458         6.737422
min     1.000000     0.000000             4.000000        20.000000
25%     9.750000     1.000000             7.552500        30.000000
50%    18.500000     2.000000            10.165000        34.500000
75%    27.250000     2.000000            14.060000        40.000000
max    36.000000     5.000000            23.300000        48.000000

### Perguntas:

1) o que faz o código acima (dados.describe())?

2) Quais informações são apresentadas?

3) Todas as variáveis têm o mesmo número de informações?

4) O que significa a variável "mean" e "std"?

5) Considerando o o máximo, o mínimo, a média e o "std", o que podemos inferir sobre as variáveis nº de filhos, salário e idade?

6) Por que não aparecem as variáveis "Grau de instrução" e "Região de Procedência"?

7) Considerando todas as análises e informações coletadas, a média é um bom indicativo para os dados?

8) Calcule a mediana e a moda dos dados apresentados na saída dados.describe(). Os valores encontrados são próximos à média?

1) Descreve as variáveis numéricas da tabela em medidas

2) São apresentados conteúdos resumos dessas variáveis

3) Para a categoria numérica, sim.

4) Média e o Desvio padrão

5) 1.268028,	4.587458,	6.737422 são as médias respctivas. Podemos inferiir que nessa ordem temos um crescente dispersão nos seus dados

6) Por que a descrição tem orientação específica a medidas resumos, o que segrega os tipos de variáveis.

7) Depende, sozinha não.

8) 2.000000,	10.165000,	34.500000 são as medianas como exposto em "50%". Dados as dimensões, Número de filhos e idades está relativamente perto, no entanto, a de salário não.